Add DNSCrypt support for dnsdist

rgacogne opened the pull request:
The support is disabled by default and can be enabled
with --enable-dnscrypt.
Creating certificates and keys is supported, as well
as basic unit tests.

pieterlexis commented on pdns/dnsdist.cc:
```diff
@@ -927,18 +1012,18 @@ try
+      cout<<"-g,--gid gid                Change the process group ID after binding sockets\n";
+      cout<<"-h,--help                   Display this helpful message\n";
+      cout<<"-l,--local address          Listen on this local address\n";
+      cout<<"-s,--supervised             Don't open a console, I'm supervised\n";
```
`-s` isn't hooked up in the `getopt_long`-call on line 990 as I wanted people to consciously set this option.

zeha replied:
... and most users probably won't ever set it (manually).

rgacogne commented:
Reverted the `-s` modification, which had nothing to do there anyway. Fixed the unwarranted tabs vs white-spaces changes. Removed the comment about the spec being wrong, it has been fixed.
